### ISSUE TITLE
event trigger parameters

### ISSUE THREAD

**Issue by Charles546** [honeydipper/honeydipper#108]
When defining an event trigger, we can specify the payload matching criteria, but we don't have a way to specify a parameter.  For example, for pubsub event trigger, we may be able to pass the project, topic name, subscription name as matching criteria, but, it is not suitable to pass a service account as a matching criteria too, although it is doable.

```
rules:
  - when:
      driver: gcloud-pubsub
      conditions:
        project: foo
        topic: bar
        service_account: ENC[blah...]
    do:
      ...
```
The `service_account` is required for the driver to establish the connection but it is not part of match criteria and should not be used as such.  Below is a possible approach.

```
rules:
  - when:
      driver: gcloud-pubsub
      conditions:
        project: foo
        topic: bar
      parameters:
        service_account: ENC[blah...]
    do:
      ...
```

We need a way to specify the parameters needed for the driver to listen to the events, and the parameters don't need to be used for comparing against event payload. 

This is needed to complete #89

**Comment by Charles546** [comment]
Just did an assessment on how much work this issue involves. Looks like it is not trivial, and potentially the changes will conflict with another branch where the DipperCL is being re-engineered. I am going to delay this and eventually merge this with the effort of DipperCL re-engineering.

For now, we can continue implementing the `gcloud-pubsub` driver by not allowing service account in the `conditions` but taking the service account from the global driver data.

```
drivers:
  gcloud-pubsub:
    service_account: ENC[blah...]
...
```

**Comment by mc422** [comment]
@Charles546 if we have a conclusion on where to put `service_account` in driver, I can refactor my code in #103 to read service_account from global driver data.

**Comment by Charles546** [comment]
This is released in 1.0.0
